src/osd/PGBackend.cc

```diff
@@ -409,7 +409,8 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
   if (auth.size != candidate.size) {
     if (error != CLEAN)
       errorstream << ", ";
-    error = SHALLOW_ERROR;
+    if (error != DEEP_ERROR)
+      error = SHALLOW_ERROR;
```

Comment:
There is a comment above that says that the shallow error here takes precedence because this can be seen by both kinds of scrubs.  This is a bit odd, but I'd say remove this entire commit unless you are seeing a particular problem.